src/components/graph/graph.coords.js

```diff
@@ -0,0 +1,333 @@
+import CONST from "./graph.const";
+import { logWarning } from "../../utils";
+
+/**
+ * Computes the normalized vector from a vector.
+ * @param {Object} vector a 2D vector with x and y components
+ * @param {number} vector.x x coordinate
+ * @param {number} vector.y y coordinate
+ * @returns {Object} normalized vector
+ * @memberof Graph/helper
+ */
+function normalize(vector) {
+  const norm = Math.sqrt(Math.pow(vector.x, 2) + Math.pow(vector.y, 2));
+  return norm === 0 ? vector : { x: vector.x / norm, y: vector.y / norm };
+}
+
+/**
+ * Calculate the length of a vector, from the center of a rectangle,
```

Comment:
Calculates the vector length from the center of a rectangle to the closest edge following a direction.